documentation/migration-guides/Az.12.0.0.0-migration-guide.md

```diff
@@ -0,0 +1,2762 @@
+```
+#### After
+```powershell
+Get-AzVmss -ResourceGroupName "" will return error for empty string validation in the parameter 
+```
+
+
+### `New-AzGalleryImageDefinition`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Starting in May 2024 the "New-AzGalleryImage" cmdlet will deploy with the Trusted Launch configuration by default. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch
+  - This change is expected to take effect from Az.Compute version: 8.0.0 and Az version: 12.0.0
+
+#### Before
+```powershell
+New-AzGalleryImageDefinition -ResourceGroupName $rgName -GalleryName $galleryName -Name $galleryImageDefinitionName -Location $location -Publisher $publisherName -Offer $offerName -Sku $skuName -OsState "Specialized" -OsType "Linux"                   defaulted to HyperVGeneration Gen1 and SecurityType Standard  in the service side 
+```
+#### After
+```powershell
+Will default to HyperVGeneration Gen2 and SecurityType TrustedLaunchSupported by code in service side 
+```
+
+
+### `New-AzVM`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Starting in May 2024 the "New-AzVM" cmdlet will deploy with the image 'Windows Server 2022 Azure Edition' by default. This will make migrating to Trusted Launch easier in the future. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch
+  - This change is expected to take effect from Az.Compute version: 8.0.0 and Az version: 12.0.0
+
+#### Before
+```powershell
+$vm = New-AzVM -ResourceGroupName $rgname -Name $vmname -Credential $cred -SecurityType "Standard" -DomainNameLabel $domainNameLabel; Creates a VM with the image Windows 2016 Datacenter. 
+```
+#### After
+```powershell
+$vm = New-AzVM -ResourceGroupName $rgname -Name $vmname -Credential $cred -SecurityType $securityTypeST -DomainNameLabel $domainNameLabel; Now generates with the Windows 2022 Azure Edition image.
+```
+
+
+### `New-AzVmss`
+
+- Cmdlet breaking-change will happen to all parameter sets
+  - Starting in May 2024 the "New-AzVmss" cmdlet will deploy with the image 'Windows Server 2022 Azure Edition' by default. This will make migrating to Trusted Launch easier in the future. To know more about Trusted Launch, please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch
+  - This change is expected to take effect from Az.Compute version: 8.0.0 and Az version: 12.0.0
+
+#### Before
+```powershell
+$vmss = New-AzVmss -ResourceGroupName $rgname -Credential $cred -VMScaleSetName $vmssName -SecurityType "Standard" -DomainNameLabel $domainNameLabel1; would create a VMSS with a Windows 2016 Datacenter image. 
+```
+#### After
+```powershell
+$vmss = New-AzVmss -ResourceGroupName $rgname -Credential $cred -VMScaleSetName $vmssName -SecurityType "Standard" -DomainNameLabel $domainNameLabel1; Now it will generate with a Windows 2022 Azure Edition image.
```

Comment:
```suggestion
$vmss = New-AzVmss -ResourceGroupName $rgname -Credential $cred -VMScaleSetName $vmssName -SecurityType "Standard" -DomainNameLabel $domainNameLabel1; 
#Now it will generate with a Windows 2022 Azure Edition image.
```